1) Support for new frameworks and bug fixes

anurag1408 opened the pull request:
1) Support for Appium python
2) bug fix for pulling the device farm icon
3) Fix passing filter tags for UiAutomator
4) Fix passing filer tags for Instrumentation
5) Add support for passing calabash profile

ahawker commented on src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java:
```diff
@@ -86,6 +94,9 @@
     // XCTest
     public String xctestArtifact;
 
+    // XCTest
```
*XCTest UI (XCUI)

ahawker commented on src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java:
```diff
@@ -708,6 +771,15 @@ private boolean validateTestConfiguration() {
+            
+            case XCTEST_UI: {
+                if (xctestUiArtifact == null || xctestUiArtifact.isEmpty()) {
+                    writeToLog("XC UI tests artifact must be set.");
```
*XC UI -> XCTest UI

ahawker commented on src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmUploadType.java:
```diff
@@ -11,11 +11,13 @@
     IOS_APP ("IOS_APP"),                                    // App
     APPIUM_JAVA_TESTNG("APPIUM_JAVA_TESTNG_TEST_PACKAGE"),  // Test
     APPIUM_JAVA_JUNIT("APPIUM_JAVA_JUNIT_TEST_PACKAGE"),    // Test
+    APPIUM_PYTHON("APPIUM_PYTHON_TEST_PACKAGE"),  // Test
```
Fix comment alignment.

ahawker commented:
I appears you're missing a "doCheckXCTestUI" style method for form validation.

ahawker commented:
Does your log changes address the same problem addressed in #30 ?

anurag1408 commented:
>> I appears you're missing a "doCheckXCTestUI" style method for form validation. 
Added UI validation for both XCTest and XCTest UI

>> Does your log changes address the same problem addressed in #30 ?
Yes

Also incorporated your other review comments
